### DISCUSSION TITLE
[公益]主页市场OSS 开放服务

### DISCUSSION THREAD

**Discussion by Joker2184** [Meloong-Git/PCL#5058] [自定义主页]
![LogoOSSProject](https://github.com/user-attachments/assets/ece2cb7f-0258-4dc3-94a9-14577e4e5e37)

为了解决 https://github.com/Hex-Dragon/PCL2/discussions/5014 所诉的问题

我们上线了 主页市场 OSS 

* 国内稳定的服务商 [✔]
* 无加速器/魔法情况下的良好连接速度 [✔]
* 简单，方便的服务管理 [✔]

这是一个 **免费公益** 的主页托管服务 由 @JingHai-Lingyun & @Joker2184 管理 @Joker2184 独立赞助 @Medicineking 提供服务器

我们提供两种托管方式：

A .Homeplaza-Of-PCL2 Alist OSS 面向普通开发者的自动+手动部署操作面板

自动化部署教程：https://github.com/HomePlaza-Of-PCL2/AlistWebDAV

B.  基于 Docker 的 在线构建服务 [需要提供Docker镜像]（HomepageBuilder [✔] NodeJS[等待支持]）

你可以通过加入群 719457352 获取服务的最新资讯

具体信息https://github.com/orgs/HomePlaza-Of-PCL2/discussions/2

你可以通过测试链接
```
http://pclhomeplazaoss.lingyunawa.top:26994/d/Homepages/Joker2184/UpdateHomepage.xaml 
```
来测试 OSS 链接是否正常

测速：
![QQ截图20241104190508](https://github.com/user-attachments/assets/8650cd19-ac33-4dbb-9c36-1f3ea82dfdc6)

![QQ截图20241104190515](https://github.com/user-attachments/assets/ee37be7f-3fb3-428d-be8b-bd9e981c353c)

**Comment by LTCatt** [comment]
如果需要我这换链接的话艾特一下我就行（

**Reply by MoYuan-CN** [reply]
在过一阵子，测试测试稳定性先

**Comment by wuliaodexiaoluo** [comment]
cf workers 是 ts
这需要再拉一个 nodejs 的容器的说

**Reply by Joker2184** [reply]
> nodejs

abbb（行 周末看看云凌怎么说

**Comment by wuliaodexiaoluo** [comment]
<details>
<summary>效果图</summary>

![image](https://github.com/user-attachments/assets/33a061da-5957-49e8-94f7-f2575320cd6a)

</details>

试搓了一下，感觉安全风险很大（（（

**Reply by wuliaodexiaoluo** [reply]
目前既有的安全策略：
执行构建脚本时自动切换到低权限账户构建和执行

**Comment by z0z0r4** [comment]
基本上没看出哪一步是跳转到 OSS 了，真的吗

```
C:\Users\DELL>curl -L -k -H "User-Agent: PCL2/homepage_collection" -H "Referer: homepage_collection.pcl2.server" "https://pclhomeplazaoss.lingyunawa.top:27000" -v
* Host pclhomeplazaoss.lingyunawa.top:27000 was resolved.
* IPv6: (none)
* IPv4: 27.218.136.111
*   Trying 27.218.136.111:27000...
* Connected to pclhomeplazaoss.lingyunawa.top (27.218.136.111) port 27000
* schannel: disabled automatic use of client certificate
* ALPN: curl offers http/1.1
* ALPN: server accepted http/1.1
* using HTTP/1.x
> GET / HTTP/1.1
> Host: pclhomeplazaoss.lingyunawa.top:27000
> Accept: */*
> User-Agent: PCL2/homepage_collection
> Referer: homepage_collection.pcl2.server
>
* Request completely sent off
* schannel: remote party requests renegotiation
* schannel: renegotiating SSL/TLS connection
* schannel: SSL/TLS connection renegotiated
* schannel: remote party requests renegotiation
* schannel: renegotiating SSL/TLS connection
* schannel: SSL/TLS connection renegotiated
< HTTP/1.1 200 OK
< Date: Thu, 28 Nov 2024 07:08:39 GMT
< Content-Type: text/html
< Transfer-Encoding: chunked
< Connection: keep-alive
< Set-Cookie:  sl-session=ycSnQXdoSWcXf60EM/L0Kg==; SameSite=None; Secure; Path=/; Max-Age=86400; HttpOnly
<

**Reply by Joker2184** [reply]
本质就一个alist下载（

**Reply by Joker2184** [reply]
> 基本上没看出哪一步是跳转到 OSS 了，真的吗
> 
> ```
> C:\Users\DELL>curl -L -k -H "User-Agent: PCL2/homepage_collection" -H "Referer: homepage_collection.pcl2.server" "https://pclhomeplazaoss.lingyunawa.top:27000" -v
> * Host pclhomeplazaoss.lingyunawa.top:27000 was resolved.
> * IPv6: (none)
> * IPv4: 27.218.136.111
> *   Trying 27.218.136.111:27000...
> * Connected to pclhomeplazaoss.lingyunawa.top (27.218.136.111) port 27000
> * schannel: disabled automatic use of client certificate
> * ALPN: curl offers http/1.1
> * ALPN: server accepted http/1.1
> * using HTTP/1.x
> > GET / HTTP/1.1
> > Host: pclhomeplazaoss.lingyunawa.top:27000
> > Accept: */*
> > User-Agent: PCL2/homepage_collection
> > Referer: homepage_collection.pcl2.server
> >
> * Request completely sent off
> * schannel: remote party requests renegotiation
> * schannel: renegotiating SSL/TLS connection
> * schannel: SSL/TLS connection renegotiated
> * schannel: remote party requests renegotiation
> * schannel: renegotiating SSL/TLS connection
> * schannel: SSL/TLS connection renegotiated
> < HTTP/1.1 200 OK
> < Date: Thu, 28 Nov 2024 07:08:39 GMT
> < Content-Type: text/html
> < Transfer-Encoding: chunked
> < Connection: keep-alive
> < Set-Cookie:  sl-session=ycSnQXdoSWcXf60EM/L0Kg==; SameSite=None; Secure; Path=/; Max-Age=86400; HttpOnly
> <
> ```

如果有更好的OSS服务也可以推荐看看.jpg Minio不能设置长时间下载nnd（

**Reply by wuliaodexiaoluo** [reply]
>如果有更好的OSS服务也可以推荐看看.jpg Minio不能设置长时间下载nnd（

Huh?
没理解啥意思

**Reply by wuliaodexiaoluo** [reply]
不过既然是 Alist
也就是可以玩点好玩的

Like [HomePlaza-Of-PCL2/discussions#3](https://github.com/orgs/HomePlaza-Of-PCL2/discussions/3)

**Reply by Joker2184** [reply]
> > 如果有更好的OSS服务也可以推荐看看.jpg Minio不能设置长时间下载nnd（
> 
> Huh? 没理解啥意思

就是（ 之前一开始打算MINIO的（可以自己搜搜（
比Alist好（
问题是 无法生成长时间的下载链接（最多7天好像（

最后发现Alist的链接PCL能用就上Alist了

**Reply by z0z0r4** [reply]
所以这不是 OSS，是 alist 本地下载…？

**Reply by Joker2184** [reply]
> 所以这不是 OSS，是 alist 本地下载…？

你说得对（
是网盘.jpg

**Reply by LinQingYuu** [reply]
>就是（ 之前一开始打算MINIO的（可以自己搜搜（
比Alist好（
问题是 无法生成长时间的下载链接（最多7天好像（

持久链接的话，根据官网文档，需要将存储桶设置为开放
但是 MinIO 本身似乎不支持直接修改配置，只能通过官方客户端连接服务器，由客户端对服务端进行调试，进而修改服务端配置

~这不就是 adb 么~

**Reply by wuliaodexiaoluo** [reply]
>但是 MinIO 本身似乎不支持直接修改配置，只能通过官方客户端连接服务器，由客户端对服务端进行调试，进而修改服务端配置

我勒个 adb 啊....
周末看看怎么折腾

**Reply by Joker2184** [reply]
> > 但是 MinIO 本身似乎不支持直接修改配置，只能通过官方客户端连接服务器，由客户端对服务端进行调试，进而修改服务端配置
> 
> 我勒个 adb 啊.... 周末看看怎么折腾

周末换minio？如何

**Reply by LinQingYuu** [reply]
>周末换minio？如何

周六再说，学校似乎没打算让我们今天回去（（（

EDIT:同学关系，不是小号

**Reply by Joker2184** [reply]
o
